src/hera/workflows/_mixins.py

```diff
@@ -621,12 +621,17 @@ def _build_arguments(self) -> Optional[ModelArguments]:
         for arg in self.arguments:
             if isinstance(arg, dict):
                 for k, v in arg.items():
-                    value = Parameter(name=k, value=v)
-                    result.parameters = (
-                        [value.as_argument()]
-                        if result.parameters is None
-                        else result.parameters + [value.as_argument()]
-                    )
+                    if isinstance(v, Parameter):
+                        value = v.as_argument()
+                    elif isinstance(v, ModelParameter):
+                        value = Parameter.from_model(v).as_argument()
+                    else:
+                        value = v
```

Comment:
We would still need to create a parameter from the given dict value `v` as `result.parameter` is a list of argument parameters (this is what broke CI lol).

```suggestion
                        value = Parameter(name=k, value=v).as_argument()
```

We could call `as_argument` after setting `value` in each of the three `if` branches, not sure if it's more readable but it's the common action after acquiring a Hera `Parameter`.